### ISSUE TITLE
populate's network detection interferes with role assigment

### ISSUE THREAD

**Issue by jan--f** [SUSE/DeepSea#120]
I want to deploy ceph on an openstack cluster where all nodes have their own /32 network. Routing is managed elsewhere. The proposal runner (rightly) complains about this network situation:
```
File "/srv/modules/runners/populate.py", line 704, in public_cluster
   raise ValueError("No network exists on at least 4 nodes")
No network exists on at least 4 nodes
```
The plan was to run stage 2 and then fill in the public_interface addresses in the ```/srv/pillar/ceph/stack``` tree. However the proposal runner doesn't create any files in the role-mon directory. Stage 2 complains about this:
```
role-mon/cluster/target158069079237.sls matched no files
role-mon/stack/default/ceph/minions/target158069079237.yml
```

Stage 3 then of course complains about the absence of monitors.
I think the proposal runner should complain about the networks, but still create at the role assignments under ```role-mon/cluster```.

**Comment by jan--f** [comment]
```/srv/pillar/ceph/proposals/config/stack/default/ceph/cluster.yml``` and ```/srv/pillar/ceph/proposals/role-admin/cluster/``` are also not created in this setup (for the mon nodes in the latter case).

**Comment by rjfd** [comment]
@jan--f this problem was the use case that I used for creating the PR #96

**Comment by jan--f** [comment]
@rjfd ah ok. I'll give your code a test run.

**Comment by jan--f** [comment]
Fixed-by: #96
